File: domain/src/main/kotlin/com/schedule/shift/domain/model/ScheduleWeek.kt
package com.schedule.shift.domain.model

import java.time.LocalDate

data class ScheduleWeek(
    val weekStartDate: LocalDate,
    val days: List<ScheduleDay>,
) {
    init {
        require(days.size == DAYS_IN_WEEK) {
            "ScheduleWeek must have exactly $DAYS_IN_WEEK days, got ${days.size}"
        }
    }

    companion object {
        const val DAYS_IN_WEEK = 7
    }
}

File: ios/shift/shift/Domain/ScheduleWeek.swift
import Foundation

struct ScheduleWeek: Hashable, Sendable {
    static let daysInWeek = 7

    let weekStartDate: Date
    let days: [ScheduleDay]

    init(weekStartDate: Date, days: [ScheduleDay]) {
        precondition(days.count == Self.daysInWeek, "ScheduleWeek must have exactly 7 days")
        self.weekStartDate = weekStartDate
        self.days = days
    }
}
